HopsanGUI/GUIObjects/GUIComponent.cpp

```diff
@@ -139,8 +139,60 @@ QString Component::getTypeCQS() const
 //! @brief Set a parameter value to be mapped to a System parameter
 bool Component::setParameterValue(QString name, QString value, bool force)
 {
+    if(this->getTypeName() == "FMIWrapper" && name == "path") {
```

Comment:
I'm thinking of a more general solution where a component can have different `<configuration>` tags in the CAF file. Then it would be possible to switch between them using a drop-down menu. This could example change a pump between fixed and variable, or enable/disable friction/leakage etc. But the FMIWrapper is still a special case, since we can't know in advance which parameters and ports we want. We need to first parse the FMI model description and then auto place the ports.

Reply:
Reconfiguration would need to be related to the core somehow, so that it works in non GUI applications.
IS the "reconfigure" parameter a special parameter, in that case we could add it as a new parameter type. (then we would get automatic "tagging" that the parameter would need to trigger reconfigure. Then if the CAF have multiple configuration options listed, then those only represent the graphics (port placements and such) for the cases already implemented inside the component and selectable from this "special parameter" similar to the one called "conditional" now ?

Reply:
Reconfiguration can be triggered by different type of parameters I would say. It could be a string for a FMU file, an enumerable for changing between configurations or an integer for choosing e.g. number of outputs. We will need a tag in the general parameter class I think, with some `doesTriggerReconfiguration()` function.

Reply:
so what to do in this case, it is still hard coded for `this->getTypeName() == "FMIWrapper" && name == "path"`

Reply:
I think we can keep the component-specific code for now. I guess the GUI could check whether or not a reconfiguration was triggered and update ports if it was, but do we always want ports to be placed in the same way for all reconfigurable components? For some components reconfiguration should probably turn on or off a port, with a fixed position specified in the CAF file. It should even be possible to change the component icon depending on the reconfiguration, e.g. changing a pump from fixed to variable.